StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/MemberOrderHelper.cs

```diff
@@ -0,0 +1,169 @@
+        /// <param name="member">The member to wrap.</param>
+        public MemberOrderHelper(MemberDeclarationSyntax member)
+        {
+            this.Member = member;
+            var modifiers = member.GetModifiers();
+
+            this.modifierFlags = GetModifierFlags(modifiers);
+            this.accessibilty = GetAccessibilityFlags(modifiers);
+        }
+
+        private enum AccesibilityFlags
+        {
+            /// <summary>
+            /// Private accessibility
+            /// </summary>
+            Private = 1,
+
+            /// <summary>
+            /// No accessibility specified
+            /// </summary>
+            None = 1 << 1,
+
+            /// <summary>
+            /// Protected accessibility
+            /// </summary>
+            Protected = 1 << 2,
+
+            /// <summary>
+            /// Protected Internal accessibility
+            /// </summary>
+            ProtectedInternal = 1 << 3,
+
+            /// <summary>
+            /// Internal accessibility
+            /// </summary>
+            Internal = 1 << 4,
+
+            /// <summary>
+            /// Public accessibility
+            /// </summary>
+            Public = 1 << 5,
+        }
+
+        [Flags]
+        private enum ModifierFlags
+        {
+            /// <summary>
+            /// No modifiers
+            /// </summary>
+            None = 0,
+            /// <summary>
+            /// Readonly modifier
+            /// </summary>
+            Readonly = 1,
+            /// <summary>
+            /// Static modifier
+            /// </summary>
+            Static = 1 << 2,
+            /// <summary>
+            /// Const modifier
+            /// </summary>
+            Const = 1 << 3,
+        }
+
+        /// <summary>
+        /// The wrapped member.
+        /// </summary>
+        /// <value>
+        /// The wrapped member.
+        /// </value>
+        public MemberDeclarationSyntax Member { get; }
+
+        /// <summary>
+        /// The priority for this member.
+        /// </summary>
+        /// <value>
+        /// The priority for this member.
+        /// </value>
+        public int Priority => (int)this.modifierFlags + ((int)this.accessibilty * 100/*the * 100 ensures the accesibility is more important than the modifier*/);
+
+        /// <summary>
+        /// The priority for this member only from accesibility.
+        /// </summary>
+        /// <value>
+        /// The priority for this member.
+        /// </value>
+        public int AccessibilityPriority => (int)this.accessibilty;
+
+        /// <summary>
+        /// The priority for this member only from modifiers.
+        /// </summary>
+        /// <value>
+        /// The priority for this member.
+        /// </value>
+        public int ModifierPriority => (int)this.modifierFlags;
+
+        private static AccesibilityFlags GetAccessibilityFlags(SyntaxTokenList syntax)
+        {
+            if (syntax.Any(SyntaxKind.PublicKeyword))
```

Comment:
:blub: Use `StyleCop.Analyzers.Helpers.AccessLevelHelper.GetAccessLevel(member). In the case of `AccessLevel.NotSpecified` return `AccessLevel.Private` instead. The code below contains a bug that will not properly handle `protected internal`.